packages/sdk/src/protocols/filter.ts

```diff
@@ -193,6 +213,38 @@ export class SubscriptionManager {
       log.info(`${type} successful for all peers`);
     }
   }
+
+  private startKeepAlivePings(interval: number): void {
+    if (this.keepAliveTimer) {
+      log.info("Recurring pings already set up.");
+      return;
+    }
+
+    this.keepAliveTimer = window.setInterval(() => {
```

Comment:
good point, the reason why I went with `window` is to make typescript understand that it is `browser` API, otherwise I need to suppress types 

making it using API directly with type trick

Reply:
> otherwise I need to suppress types

the types natively exist in TypeScript -- no need to supress types

Reply:
there is an ambiguity in types 

in `nodejs` it returns `NodeJS.Timeout` - https://github.com/DefinitelyTyped/DefinitelyTyped/blob/778ad7a80e9876f258270d84d1db4e6e1d5e3796/types/node/timers.d.ts#L191

where as in browser it is a number 

hence the need to do `as number` to help TS determine it is browser interface